src/Xrefcheck/Core.hs

```diff
@@ -77,8 +75,82 @@ data Reference = Reference
   , rAnchor :: Maybe Text
     -- ^ Section or custom anchor tag.
   , rPos    :: Position
+    -- ^ More info about the link.
+  , rInfo   :: ReferenceInfo
```

Comment:
I keep it because of the draft that is proposed in the issue, but in fact I thought that it was proposed in this way to avoid repeated calculations. This value can be always calculated again with just other stored data (rLink), would you prefer to not store it?

Reply:
Ah right, I completely forgot about the suggestion made in the issue, thanks for the clarification.

Good then, let's store it for sure.